### PR TITLE
feat: no-restricted-vi-method rule

### PR DESCRIPTION
**Pull request by BrandonJHoff**
This PR fixes/implements the following **bugs/features**

* [ ] Bug
* [x] Feature
* [ ] Breaking changes

**Closing issues**

Fixes #

### CODE REVIEW

File: src/rules/no-restricted-vi-methods.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+				errors: [{
+					messageId: 'restrictedViMethod',
+					data: {
+						message: null,
+						restriction: 'fn'
+					},
+					column: 4,
+					line: 1
+				}]
+			},
```
Comment (BrandonJHoff) [inline src/rules/no-restricted-vi-methods.test.ts:38]: I tried adding the following test, but it kept failing (it wouldn't find any errors). I'm not sure why this test doesn't work:

```
{
	code: 'vi["fn"]()',
	options: [{ fn: null }],
	errors: [{
		messageId: 'restrictedViMethod',
		data: {
			message: null,
			restriction: 'fn'
		},
		column: 4,
		line: 1
	}]
}
```

**Comment by veritem** [comment]
There is a bug in `parseVitestFnCall(node, context)` function.

**Comment by BrandonJHoff** [comment]
> There is a bug in `parseVitestFnCall(node, context)` function. I am working on addressing it, but if you have time, that's the right place to look trouble shooting.

I believe I found the issue and opened a different PR with a fix. -- https://github.com/veritem/eslint-plugin-vitest/pull/55

**Review by veritem — APPROVED** [review]
Thank you!!!!

**Pull request merged**